app/api/business/admin_download_report_business.py

```diff
@@ -0,0 +1,33 @@
+from app.api.services import (
+    briefs,
+    suppliers,
+    domain_service
+)
+
+from app.api.admin_csv import (
+    generate_all_sellers_category_max_daily_rates,
+    generate_all_selers_approved_in_category
+)
+
+
+def get_result(current_user, report_type):
+    csv_generator = None
+    result = None
+    report_file_name = None
+    domains = domain_service.get_active_domains()
+    domain_list = []
+    for domain in domains:
+        domain_list.append(domain.name)
+
+    if report_type == 'allSellersCategoryMaxDailyRates':
+        result = suppliers.get_all_suppliers()
+        csv_generator = generate_all_sellers_category_max_daily_rates
+        report_file_name = 'all-sellers-category-max-daily-rates.csv'
+
+    else:
+        # report tyoe allSellersApprovedInCategory here
```

Comment:
I think we should check the report type here.  You may want to raise an error if an unknown report type is passed to this function.

Reply:
Not sure if required exception. report_type  is already checked at line 22.  Here we assume it will be all-sellers-approved-in-category

Reply:
better to check report_type at request level instead on business level. See admin_donwload_report_business.py